File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/controller/HttpApiResourceRemoteApiController.java
package com.jmsoftware.maf.springcloudstarter.controller;

import cn.hutool.core.collection.CollUtil;
import com.jmsoftware.maf.common.bean.ResponseBodyBean;
import com.jmsoftware.maf.common.domain.springbootstarter.HttpApiResourcesResponse;
import com.jmsoftware.maf.springcloudstarter.helper.HttpApiScanHelper;
import com.jmsoftware.maf.springcloudstarter.property.MafConfigurationProperties;
import lombok.RequiredArgsConstructor;
import lombok.val;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Objects;
import java.util.Optional;

/**
 * Description: HttpApiResourceRemoteApiController, change description here.
 *
 * @author 钟俊（zhongjun）, email: zhongjun@toguide.cn, date: 12/25/2020 4:33 PM
 **/
@RestController
@RequiredArgsConstructor
public class HttpApiResourceRemoteApiController {
    private final MafConfigurationProperties mafConfigurationProperties;
    private final HttpApiScanHelper httpApiScanHelper;

    @GetMapping("/http-api-resources")
    public ResponseBodyBean<HttpApiResourcesResponse> getHttpApiResource() {
        val handlerMethodMap = this.httpApiScanHelper.scan(this.mafConfigurationProperties.getIncludedPackageForHttpApiScan());
        HttpApiResourcesResponse response = new HttpApiResourcesResponse();
        handlerMethodMap.forEach((requestMappingInfo, handlerMethod) -> {
            val requestMethod = Optional.ofNullable(
                    new ArrayList<>(requestMappingInfo.getMethodsCondition().getMethods()).get(0))
                    .orElseThrow(() -> new RuntimeException("Request method mustn't be null!"));
            val httpApiResource = new HttpApiResourcesResponse.HttpApiResource();
            httpApiResource.setMethod(requestMethod);
            val patterns = Objects.requireNonNull(requestMappingInfo.getPatternsCondition()).getPatterns();
            if (CollUtil.isNotEmpty(patterns)) {
                httpApiResource.setUrlPattern(new LinkedList<>(patterns).get(0));
            } else {
                httpApiResource.setUrlPattern("/url-pattern-not-found");
            }
            response.getList().add(httpApiResource);
        });
        return ResponseBodyBean.ofSuccess(response);
    }
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/controller/HttpApiResourceRemoteApiController.kt
package com.jmsoftware.maf.springcloudstarter.controller

import com.jmsoftware.maf.common.bean.ResponseBodyBean
import com.jmsoftware.maf.common.domain.springbootstarter.HttpApiResourcesResponse
import com.jmsoftware.maf.common.domain.springbootstarter.HttpApiResourcesResponse.HttpApiResource
import com.jmsoftware.maf.springcloudstarter.helper.HttpApiScanHelper
import com.jmsoftware.maf.springcloudstarter.property.MafConfigurationProperties
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.method.HandlerMethod
import org.springframework.web.servlet.mvc.method.RequestMappingInfo
import java.util.*

/**
 * # HttpApiResourceRemoteApiController
 *
 * Description: HttpApiResourceRemoteApiController, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/13/22 1:45 PM
 */
@RestController
class HttpApiResourceRemoteApiController(
    private val mafConfigurationProperties: MafConfigurationProperties,
    private val httpApiScanHelper: HttpApiScanHelper
) {
    @GetMapping("/http-api-resources")
    fun httpApiResource(): ResponseBodyBean<HttpApiResourcesResponse> {
        val handlerMethodMap = httpApiScanHelper.scan(mafConfigurationProperties.includedPackageForHttpApiScan)
        val response = HttpApiResourcesResponse()
        handlerMethodMap.forEach { (requestMappingInfo: RequestMappingInfo, _: HandlerMethod) ->
            val requestMethod = ArrayList(requestMappingInfo.methodsCondition.methods)[0]
                ?: throw IllegalStateException("Request method mustn't be null!")
            val httpApiResource = HttpApiResource()
            httpApiResource.method = requestMethod
            val patterns = requestMappingInfo.patternsCondition?.patterns
            if (patterns?.isNotEmpty() == true) {
                httpApiResource.urlPattern = LinkedList(patterns)[0]
            } else {
                httpApiResource.urlPattern = "/url-pattern-not-found"
            }
            response.list.add(httpApiResource)
        }
        return ResponseBodyBean.ofSuccess(response)
    }
}
